extension-manifest-v3/scripts/download-engines/utils.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+
+import { createWriteStream } from 'fs';
+
+const supportedResourceTypes = [
+  'font',
+  'image',
+  'main_frame',
+  'media',
+  'ping',
+  'script',
+  'stylesheet',
+  'sub_frame',
+  'websocket',
+  'xmlhttprequest',
+  'other',
+];
+
+const supportedActions = ['block', 'allow', 'allowAllRequests'];
+
+export function getCompatRule(rule) {
+  const resourceTypes = rule.condition.resourceTypes?.filter((type) =>
+    supportedResourceTypes.includes(type),
+  );
+
+  if (
+    // Based on https://github.com/w3c/webextensions/issues/344#issuecomment-1430358116
+    rule.condition.regexFilter?.match(/(\{\d*,\d*\}|\{\d\}|\|)/) ||
+    !supportedActions.includes(rule.action.type) ||
+    (resourceTypes && resourceTypes.length === 0)
+  ) {
+    return null;
+  }
+
+  return {
+    priority: rule.priority,
+    ...(rule.action.type === 'allowAllRequests'
```

Comment:
should we add some unit tests for all that?

Reply:
Finally, this should be taken out from the repository to https://github.com/ghostery/adblocker-filters. We can keep it here to test out with upcoming release and then move it out.